packages/amplify-graphql-relational-transformer/src/utils.ts

```diff
@@ -193,6 +193,31 @@ export const ensureReferencesArray = (
   }
 };
 
+export const ensureReferencesBidirectionality = (
+  config: HasManyDirectiveConfiguration | BelongsToDirectiveConfiguration, // TODO: Add HasOnyDirectiveConfiguration
+): void => {
+  if (config.fields) {
+    throw new InvalidDirectiveError('fields and references cannot be used together.');
+  }
+
+  /*
+    1. find related directives:
+      - if hasMany --> belongsTo
+      - if hasOne --> belongsTo
+      - if belongsTo --> hasMany | hasOne
+    2. find matching propertyName
+      - directive.arguments[].references === config.references (order matters)
+    3. confirm matching type as sanity check
+      - field.type.name.value == config.type
+
+
+  */
+  // const related = config.relatedType;
+  // const otherSide = related.fields?.find((field) => {
+  //   const hasDirective = field?.directives?.length >= 1
+  // })
+};
+
```

Comment:
This is a shell of an implementation. Tracking pending work in feature branch PR description as:
> Add validation for relational bi-directionality of DDB references